api/v1/mongodbcommunity_types.go

```diff
@@ -497,6 +505,30 @@ func (m *MongoDBCommunity) StatefulSetReplicasThisReconciliation() int {
 	return scale.ReplicasThisReconciliation(m)
 }
 
+// GetUpdateStrategyType returns the type of RollingUpgradeStrategy that the
+// MongoDB StatefulSet should be configured with.
+func (m MongoDBCommunity) GetUpdateStrategyType() appsv1.StatefulSetUpdateStrategyType {
+	if !m.IsChangingVersion() {
+		return appsv1.RollingUpdateStatefulSetStrategyType
+	}
+	return appsv1.OnDeleteStatefulSetStrategyType
+}
+
+// IsChangingVersion returns true if an attempted version change is occurring.
+func (m MongoDBCommunity) IsChangingVersion() bool {
+	prevVersion := m.getPreviousVersion()
+	return prevVersion != "" && prevVersion != m.Spec.Version
```

Comment:
Q: Can `prevVersion` be empty (I think it could be in the case the first time an upgrade is performed) ? What should be the behaviour in that case?

Reply:
yes, if there are no annotations the expected behaviour is to return false as it will mean there is no version change in progress since n version has been deployed yet